kernel/sched.c

```diff
@@ -1,25 +1,52 @@
 #include <kernel/sched.h>
 #include <kernel/thread.h>
 
-extern const struct sched sched_rr;
-extern const struct sched sched_bitmap;
+extern unsigned long __sched_classes_start__;
+extern unsigned long __sched_classes_end__;
 
-static const struct sched *sched;
+/* static vars to repesent sched classes list */
+static struct sched **sched_classes = 
+    (struct sched **) &__sched_classes_start__;
+static struct sched **sched_classes_end = 
+    (struct sched **) &__sched_classes_end__;
 
-int sched_select(int sched_type)
+static struct sched *sched;
+
+int sched_init()
+{
+    int ret = 0;
+
+    /* Iteratively traverse hook list and initialize
+     * specified sched internal structure.
+     */
+    for (struct sched **clses = sched_classes;
```

Comment:
What's `clses`? You can say "class" or "clazz", which avoid C++ compilation warnings.